### ISSUE TITLE
Neo4j 2.3.1 Enterprise: Adding a label index on a property containing more than 16kb of data results in inconsistent online backups...

### ISSUE THREAD

**Issue by mikesname** [neo4j/neo4j#6207]
Running the [attached Cypher script](https://github.com/neo4j/neo4j/files/88890/test.cypher.txt) on a fresh database (via `neo4-shell`) results in the consistency problems shown below when attempting an online backup.

Removing one character from the data string so it's less than 16Kb makes the issue go away.

For reference, the test Cypher is just this:

```
CREATE (n:Item { data: '<BIG 16Kb BLOB>');

CREATE INDEX ON :Item(data);
```

I recognise that attempting to index big blobs like this isn't ideal, but I've tried to simplify the issue here from a corner case I ran into doing some experiments with the Enterprise edition.

The inconsistency report is [here](https://github.com/neo4j/neo4j/files/88898/inconsistencies-2016-01-13.13.15.45.report.txt).

Oracle Java version 1.8.0_45 on Ubuntu 15.10 x86_64.

``` bash
neo4j-enterprise-2.3.1/bin/neo4j-backup  -from single://localhost:6362 -to backup.graph.db
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=512m; support was removed in 8.0
Performing backup from 'localhost:6362'
2016-01-13 13:15:43.846+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.relationshipstore.db
2016-01-13 13:15:43.850+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.relationshipstore.db 0.00 B
2016-01-13 13:15:43.850+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.labeltokenstore.db.names
2016-01-13 13:15:43.852+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.labeltokenstore.db.names 8.00 kB
2016-01-13 13:15:43.852+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.counts.db.b
2016-01-13 13:15:43.852+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.counts.db.b 224.00 B
2016-01-13 13:15:43.852+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.propertystore.db.index.keys
2016-01-13 13:15:43.853+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.propertystore.db.index.keys 8.00 kB
2016-01-13 13:15:43.853+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.nodestore.db
2016-01-13 13:15:43.854+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.nodestore.db 8.00 kB
2016-01-13 13:15:43.854+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.relationshipgroupstore.db
2016-01-13 13:15:43.854+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.relationshipgroupstore.db 8.00 kB
2016-01-13 13:15:43.855+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.nodestore.db.labels
2016-01-13 13:15:43.855+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.nodestore.db.labels 8.00 kB
2016-01-13 13:15:43.855+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.propertystore.db
2016-01-13 13:15:43.856+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.propertystore.db 7.97 kB
2016-01-13 13:15:43.856+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.propertystore.db.index
2016-01-13 13:15:43.856+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.propertystore.db.index 8.00 kB
2016-01-13 13:15:43.856+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.relationshiptypestore.db
2016-01-13 13:15:43.857+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.relationshiptypestore.db 0.00 B
2016-01-13 13:15:43.857+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.propertystore.db.arrays
2016-01-13 13:15:43.857+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.propertystore.db.arrays 8.00 kB
2016-01-13 13:15:43.858+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.schemastore.db
2016-01-13 13:15:43.858+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.schemastore.db 8.00 kB
2016-01-13 13:15:43.859+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.propertystore.db.strings
2016-01-13 13:15:43.860+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.propertystore.db.strings 24.00 kB
2016-01-13 13:15:43.860+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.relationshiptypestore.db.names
2016-01-13 13:15:43.860+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.relationshiptypestore.db.names 8.00 kB
2016-01-13 13:15:43.860+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.labeltokenstore.db
2016-01-13 13:15:43.861+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.labeltokenstore.db 8.00 kB
2016-01-13 13:15:43.861+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore.counts.db.a
2016-01-13 13:15:43.861+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore.counts.db.a 96.00 B
2016-01-13 13:15:43.861+0000 INFO  [o.n.c.s.StoreCopyClient] Copying neostore
2016-01-13 13:15:43.862+0000 INFO  [o.n.c.s.StoreCopyClient] Copied neostore 8.00 kB
2016-01-13 13:15:43.862+0000 INFO  [o.n.c.s.StoreCopyClient] Copying schema/label/lucene/_0.cfs
2016-01-13 13:15:43.862+0000 INFO  [o.n.c.s.StoreCopyClient] Copied schema/label/lucene/_0.cfs 225.00 B
2016-01-13 13:15:43.862+0000 INFO  [o.n.c.s.StoreCopyClient] Copying schema/label/lucene/segments_1
2016-01-13 13:15:43.863+0000 INFO  [o.n.c.s.StoreCopyClient] Copied schema/label/lucene/segments_1 249.00 B
2016-01-13 13:15:43.863+0000 INFO  [o.n.c.s.StoreCopyClient] Copying schema/index/lucene/1/_0.cfs
2016-01-13 13:15:43.863+0000 INFO  [o.n.c.s.StoreCopyClient] Copied schema/index/lucene/1/_0.cfs 207.00 B
2016-01-13 13:15:43.864+0000 INFO  [o.n.c.s.StoreCopyClient] Copying schema/index/lucene/1/segments_1
2016-01-13 13:15:43.864+0000 INFO  [o.n.c.s.StoreCopyClient] Copied schema/index/lucene/1/segments_1 263.00 B
2016-01-13 13:15:43.864+0000 INFO  [o.n.c.s.StoreCopyClient] Done, copied 21 files
Full Consistency Check
.2016-01-13 13:15:45.327+0000 ERROR [o.n.c.ConsistencyCheckService] This node was not found in the expected index.
    Node[0,used=true,rel=-1,prop=0,labels=Inline(0x1000000000:[0]),light]
    Inconsistent with: IndexRule[id=1, label=0, kind=INDEX_RULE, provider={key=lucene, version=1.0}, properties=0] 8DTK2gNGWQuqdwOv9ha98O/6TYuIqO9Ku/qasXOlgf3X9FaCafzQ6kUz62EkbYJ/2fFoUd/6SkxO
65Nqwz+L6S7sjyaR3Dj/fp5mzhUe6jJy/l+tuFA2LCrjj2m2FGR6p41A7cER+SUeBJ4wT2nLPhAl

==== SNIP ===

DXw7JLEAKSGXdurzMW/pntC3QSMIEjItRzYNbVIKJ2Mm4zBUX2fqfomwlH+WsJENNeDIYL6eURuV
NEPvpsWpIX8jM/qPCwNoXXSDfPdSWu/cxFMJKz4nGvzSqSJi8c2gU2dP1aeYNEkGX4l9h9jSDyIV
TJnU3rhu76XFbZtV/LS3jxkGvRPIHif1ZAqSOrn49TSwxfDDSHNY7fFxW3m68EByFDDSUfUyXAal
JvF8+eODjd0wFA/L2X90+hLt9kid2kqxDKNtxyAR10uTDMI4+jh4UIRLMSh5
...................  10%
....................  20%
....................  30%
....................  40%
....................  50%
....................  60%
....................  70%
....................  80%
....................  90%
..................Checking node and relationship counts
....................  10%
....................  20%
....................  30%
....................  40%
....................  50%
....................  60%
....................  70%
....................  80%
....................  90%
.................... 100%
2016-01-13 13:15:45.374+0000 WARN  [o.n.c.ConsistencyCheckService] Inconsistencies found: ConsistencySummaryStatistics{
    Number of errors: 1
    Number of warnings: 0
    Number of inconsistent NODE records: 1
}
2016-01-13 13:15:45.475+0000 WARN  [o.n.c.ConsistencyCheckService] See 'backup.graph.db/inconsistencies-2016-01-13.13.15.45.report' for a detailed consistency report.
Done
WARNING: The database is inconsistent.
```

**Comment by jexp** [comment]
Thanks a lot for your feedback @mikesname 
We're looking into it.

**Comment by jexp** [comment]
Should be fixed by this PR for 3.0

https://github.com/neo4j/neo4j/pull/6213
